staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/kmsv2/v2/api.proto

```diff
@@ -28,9 +28,24 @@ message EncryptedObject {
   // KeyID is the KMS key ID used for encryption operations.
   string keyID = 2;
 
-  // EncryptedDEK is the encrypted DEK.
-  bytes encryptedDEK = 3;
+  // EncryptedDEKSource is the ciphertext of the source of the DEK used to encrypt the data stored in encryptedData.
+  // encryptedDEKSourceType defines the process of using the plaintext of this field to determine the aforementioned DEK.
+  bytes encryptedDEKSource = 3;
 
   // Annotations is additional metadata that was provided by the KMS plugin.
   map<string, bytes> annotations = 4;
+
+  // encryptedDEKSourceType defines the process of using the plaintext of encryptedDEKSource to determine the DEK.
+  EncryptedDEKSourceType encryptedDEKSourceType = 5;
+}
+
+enum EncryptedDEKSourceType {
```

Comment:
Minor: I might drop the "Encrypted" prefix. This describes a property of the unencyrpted DEKSource, it shows up in code around the unencrypted key, etc... encryptedDEKSourceType sounds fine as the field name, I'd just drop it from the enum type name.